automap/src/comm_layer/pcp_pmp_common/windows_specific.rs

```diff
@@ -0,0 +1,401 @@
+   Temporary IPv6 Address. . . . . . : 2002:aaaa:bbbb:0:cccc:eeee:cfe7:730e
+   Link-local IPv6 Address . . . . . : fe80::111:2222:3333:4444%21
+   IPv4 Address. . . . . . . . . . . : 192.168.1.28
+   Subnet Mask . . . . . . . . . . . : 255.255.255.0
+   Default Gateway . . . . . . . . . : fe80::5555:6666:7777:8888%21
+";
+        let find_routers_command = FindRoutersCommandMock::new(Ok(&route_n_output));
+
+        let result = windows_find_routers(&find_routers_command).unwrap();
+
+        assert_eq!(
+            result,
+            vec![IpAddr::from_str("fe80::5555:6666:7777:8888").unwrap()]
+        )
+    }
+
+    #[test]
+    fn find_routers_works_when_there_is_no_router_to_find() {
+        let route_n_output = "
+Windows IP Configuration
+
+   Host Name . . . . . . . . . . . . : DESKTOP-EULPUP3
+   Primary Dns Suffix  . . . . . . . :
+   Node Type . . . . . . . . . . . . : Hybrid
+   IP Routing Enabled. . . . . . . . : No
+   WINS Proxy Enabled. . . . . . . . : No
+
+Ethernet adapter Ethernet:
+
+   Connection-specific DNS Suffix  . :
+   Description . . . . . . . . . . . : Intel(R) PRO/1000 MT Desktop Adapter
+   Physical Address. . . . . . . . . : 08-00-27-4B-EB-0D
+   DHCP Enabled. . . . . . . . . . . : Yes
+   Autoconfiguration Enabled . . . . : Yes
+   Link-local IPv6 Address . . . . . : fe80::a06b:7e59:8cb5:e82f%6(Preferred)
+   IPv4 Address. . . . . . . . . . . : 10.0.2.15(Preferred)
+   Subnet Mask . . . . . . . . . . . : 255.255.255.0
+   Lease Obtained. . . . . . . . . . : Sunday, February 14, 2021 5:21:59 PM
+   Lease Expires . . . . . . . . . . : Monday, February 15, 2021 2:45:20 PM
+   Defart Gateway. . . . . . . . . . : 10.0.2.2
+   DHCP Server . . . . . . . . . . . : 10.0.2.3
+   DHCPv6 IAID . . . . . . . . . . . : 101187623
+   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-26-49-5A-82-08-00-27-4B-EB-0D
+   DNS Servers . . . . . . . . . . . : 192.168.0.1
+   NetBIOS over Tcpip. . . . . . . . : Enabled
+";
+        let find_routers_command = FindRoutersCommandMock::new(Ok(&route_n_output));
+
+        let result = windows_find_routers(&find_routers_command).unwrap();
+
+        assert_eq!(result.is_empty(), true)
+    }
+
+    #[test]
+    #[should_panic(expected = "Bad syntax from ipconfig /all")]
+    fn find_routers_works_when_ipconfig_output_cant_be_parsed() {
+        let route_n_output = "
+   Booga
+   Default Gateway. . . . . . . . . . : wibblety-poo
+   Booga
+";
+        let find_routers_command = FindRoutersCommandMock::new(Ok(&route_n_output));
+
+        let result = windows_find_routers(&find_routers_command);
+    }
+
+    #[test]
+    fn find_routers_command_handles_bad_command() {
+        let find_routers_command = FindRoutersCommandMock::new(Err("Booga!"));
+
+        let result = windows_find_routers(&find_routers_command);
+
+        assert_eq!(
+            result,
+            Err(AutomapError::ProtocolError("Booga!".to_string()))
```

Comment:
Is this really a protocol error; this rather has to do with commands for software on your device, right? Maybe a "ConfigCommandError"? But you're batter at naming things. :)